Automattic/jetpack
# My Jetpack: Add Backup 'needs-attention' status when backups are failing.

This PR adds a 'needs-addention' status to the **Backup** product card in My Jetpack **_when Backups are failing_**.
The Primary CTA changes to "Troubleshoot".

![Screen Shot on 2024-12-05 at 14-57-05](https://github.com/user-attachments/assets/c98ffca5-4988-492c-afa3-142b31cf459d)





Fixes: https://github.com/Automattic/jetpack-roadmap/issues/1938

## Proposed changes:

* Add a "Needs attention" status.
* When Jetpack is connected, and a Backup plan is owned, and when Backups are failing for whatever reason, the status of the Backup card becomes "Needs attention".
* The primary CTA becomes "Troubleshoot" and links to a Jetpack.com support post about troubleshooting Backup issues.

### Other information:


## Jetpack product discussion



Project Thread: pbNhbs-bJN-p2
Design post: p1HpG7-umC-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






**Testing notes:** To test this PR, basically we need to force a backup to fail. I tried various ways to make a backup fail, but most my attempts were unsuccessful. The only way I've been able to make a failed backup is by making the site unaccessible, and then queueing the backup to run from the MC Rewind Debugger. To make the site unaccessible, I simply killed my Jurassic Tube tunneling but there are other various ways to do this such as by stopping Apache, blocking port 80, inserting `die()` or `exit()` command in index.php, etc..
There is another promising way of making a backup fail due to a "No such file" error, but I believe the method only works on a self-hosted site that is Not running within a Docker container. (I was not able to get it to work on my local dev environment or with Jurassic Ninja (confirmed it does not work on a JN site here: p1733501055164569/1733349201.389539-slack-CS8UYNPEE )
Instructions on how to carry out this method are here: _Generating “No such file” errors during backup_: pbuNQi-1Wi-p2
If you know of any other ways to trigger a backup failure, please let me know!

### Testing steps:

* Run this PR branch locally in a [local Jetpack dev environment](https://github.com/Automattic/jetpack/blob/trunk/docs/development-environment.md#docker-supported-recommended) (via Docker & JurassicTube or Ngrok tunneling).
* Once Jetpack is up and running, ensure it is fully connected. (Also ensure you are NOT sandboxed, and NOT using STORE_SANDOX or anything like that.)
* Purchase a Backup plan (using A11N credits).
* Wait for your first full backup to complete.
* Kill your Jurassic Tube or Ngrok tunneling, so the site is no longer accessible via the tunneling domain.
* Open your site in the MC Rewind Debugger: PCYsg-lNo-p2
* Queue up a new full backup on your site.
* Wait until the full backup completes:
    - in the MC Rewind debugger page you can find the current backup in the "Recent Full Backup Attempts" section and wait for it to show that is done (it should finish with the status 'not-accessible').
    - You can also open the WPCOM API Developer Console: https://developer.wordpress.com/docs/api/console/. (sign in)
        - Check the rewind backups endpoint at `wpcom/v2`, endpoint: `/sites/239657125/rewind/backups?force=wpcom`, and verify the status of the last backup is `not-available`.
* Now re-connect your Jurassic Tube or Ngrok tunneling.
* Open your site ad go to My Jetpack page.
* Look at the backup card. The status of the backup card should be "Needs attention". The card should have a red border around it, and the CTA should say, "Troubleshoot" which points to the Jetpack.com Troubleshooting backup page. (see screenshot:)
![Screen Shot on 2024-12-07 at 22-50-30](https://github.com/user-attachments/assets/8685297d-5efc-484e-bdb3-d355821d3362)

If you've gotten this far, you way want to go ahead and review [this follow-up PR](https://github.com/Automattic/jetpack/pull/40512) too, since you've already triggered a failed backup (the PR testing instructions are almost the same).
